cmd/admin/cert.go

```diff
@@ -241,6 +242,15 @@ func (a *admin) serialsFromRegID(ctx context.Context, regID int64) ([]string, er
 	return serials, nil
 }
 
+func (a *admin) serialsFromCertPEM(_ context.Context, filename string) ([]string, error) {
+	cert, err := core.LoadCert(filename)
+	if err != nil {
```

Comment:
If  `-malformed=true` which indicates that the cert cannot be parsed, this is still going to fail on a malformed cert because `core.LoadCert` calls `x509.ParseCertificate`. What should the behavior of `-malformed` and `-cert-file` be instead?

Reply:
Honestly that seems fine to me? The point of `-malformed` is to tell the RA that it shouldn't try to parse the cert. If parsing fails before we've even sent any gRPC requests, that's a safe fail-closed.

Reply:
Is the malformed flag intended to be used only when specifying a private key?

Reply:
No, the revoke-cert subcommand revokes by by serial number: either a single one, or a list supplied by a file or an incident table.

But when a cert is revoked, the RA needs to know which intermediate issued it. The [malformed flag controls](https://github.com/letsencrypt/boulder/blob/2b8dea9821233bf4dd185e69f322f20abb8f2017/ra/ra.go#L2287) whether it determines the issued by parsing the cert vs by looking it up in the database. (This in turn controls whether we can do an akamai purge or not.)

So the malformed flag controls the RA's behavior, not the admin tool's behavior. You are right that it probably generally doesn't make sense to pass both the malformed flag and the cert-file flag at the same time. But I think the way in which it will fail if someone _does_ is just fine, and adding flag validation to catch this one case is not worthwhile.